Since these binaries are going to be setuid root, we want 
don't want to allow lazy bindings, otherwise we allow some 
information leakage.

http://bugs.gentoo.org/77526

--- iputils-s20121011/Makefile.orig	2012-11-01 17:12:14.226219387 +0100
+++ iputils-s20121011/Makefile	2012-11-01 17:15:12.589545237 +0100
@@ -42,15 +42,15 @@
 clockdiff: clockdiff.o
 	$(LINK.o) -o $@ $^ $(LIB_CAP)
 traceroute6: traceroute6.o
-	$(LINK.o) -o $@ $^ $(LIB_CAP)
+	$(LINK.o) -o $@ $^ $(LIB_CAP) -Wl,-z,now -Wl,-z,relro
 tftpd: tftpd.o tftpsubs.o
 	$(LINK.o) -o $@ $^ $(LIB_CAP)
 arping: arping.o
-	$(LINK.o) -o $@ $^ $(LIB_SYSFS) $(LIB_CAP)
+	$(LINK.o) -o $@ $^ $(LIB_SYSFS) $(LIB_CAP) -Wl,-z,now -Wl,-z,relro
 ping: ping.o ping_common.o
-	$(LINK.o) -o $@ $^ $(LIB_CAP)
+	$(LINK.o) -o $@ $^ $(LIB_CAP) -Wl,-z,now -Wl,-z,relro
 ping6: ping6.o ping_common.o
-	$(LINK.o) -o $@ $^ -lresolv -lcrypto $(LIB_CAP)
+	$(LINK.o) -o $@ $^ -lresolv -lcrypto $(LIB_CAP) -Wl,-z,now -Wl,-z,relro
 
 ping.o ping6.o ping_common.o: ping_common.h
 tftpd.o tftpsubs.o: tftp.h
